### PR TITLE
Detail the creation of a new question type

### PR DESCRIPTION
**Pull request by NicolasHairon**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no

Documentation to create new question type

### CODE REVIEW

File: plugin/exo/README.md [hunk 1]
```diff
@@ -67,15 +67,73 @@ This parameter permits to handle custom serialization logic (e.g. enable or disa
 
 Add your new type to [QuestionType.php](Library/Question/QuestionType.php).
```
Comment (Elorfin) [inline plugin/exo/README.md]: This is now `Library/Item/ItemType.php`.

File: plugin/exo/README.md [hunk 2]
```diff
@@ -67,15 +67,73 @@ This parameter permits to handle custom serialization logic (e.g. enable or disa
...
+      "type": "string"
+    }
+  }
+}
+```
+Also create tests in `test` add the question type in:
+- `test/answer-data`
+- `test/question`
+
+and reference them in [Show assert.js](assert.js).
```
Comment (Elorfin) [inline plugin/exo/README.md]: `assert.js` is not located in this repository, so this link is dead. Use `https://github.com/json-quiz/json-quiz/assert.js` instead.

File: plugin/exo/README.md [hunk 3]
```diff
@@ -67,15 +67,73 @@ This parameter permits to handle custom serialization logic (e.g. enable or disa
...
+
+### Create the MIME type
+
+Create a new MIME type in file [Show QuestionType.php](Library/Question/QuestionType.php).
```
Comment (Elorfin) [inline plugin/exo/README.md]: Now `Library/Item/ItemType.php`

File: plugin/exo/README.md [hunk 4]
```diff
@@ -67,15 +67,73 @@ This parameter permits to handle custom serialization logic (e.g. enable or disa
...
+
+### Create the data model
+
+Create a question type entity in `Entity/QuestionType`.
```
Comment (Elorfin) [inline plugin/exo/README.md]: Now `Entity/ItemType`

File: plugin/exo/README.md [hunk 5]
```diff
@@ -93,7 +151,38 @@ class ChoiceSerializer implements QuestionHandlerInterface, SerializerInterface
 }
 ```
 
-The `QuestionSerializer` is now able to forward the serialization to the correct question type serializer based 
-on the question type.
+The `QuestionSerializer` is now able to forward the serialization to the correct
```
Comment (Elorfin) [inline plugin/exo/README.md]: Now `ItemSerializer`

File: plugin/exo/README.md [hunk 6]
```diff
@@ -93,7 +151,38 @@ class ChoiceSerializer implements QuestionHandlerInterface, SerializerInterface
...
+question type serializer based on the question type.
 
 ### Create the Validator
+
+Create a question type validator in `Library/Validator/JsonSchema/Question/Type`.
```
Comment (Elorfin) [inline plugin/exo/README.md]: Now `Library/Validator/JsonSchema/Item/Type`

File: plugin/exo/README.md [hunk 7]
```diff
@@ -93,7 +151,38 @@ class ChoiceSerializer implements QuestionHandlerInterface, SerializerInterface
...
+
+A definition is all necessary references to handle a question type:
+- question Validator
+- answer Validator
+- serializer
+- method `correctAnswer(AbstractQuestion $question, $answer)` : correct a given
+answer with expected, unexpected and missing answers
+- method `expectAnswer(AbstractQuestion $question)` : gives global score
+
+Create a definition for the question type in `Library/Question/Definition`.
```
Comment (Elorfin) [inline plugin/exo/README.md]: Now `Library/Item/Definition`

File: plugin/exo/README.md [hunk 8]
```diff
@@ -93,7 +151,38 @@ class ChoiceSerializer implements QuestionHandlerInterface, SerializerInterface
...
+
+## [Font-end] Adding a new question type
```
Comment (stefk) [inline plugin/exo/README.md]: Typo (*front*)

File: plugin/exo/README.md [hunk 9]
```diff
@@ -93,7 +151,38 @@ class ChoiceSerializer implements QuestionHandlerInterface, SerializerInterface
...
+The `QuestionSerializer` is now able to forward the serialization to the correct
+question type serializer based on the question type.
 
 ### Create the Validator
+
+Create a question type validator in `Library/Validator/JsonSchema/Question/Type`.
+
+### Create the Definition
+
+A definition is all necessary references to handle a question type:
```
Comment (stefk) [inline plugin/exo/README.md]: Something wrong in this sentence.

**Pull request merged**